BUILD.gn

```diff
@@ -0,0 +1,507 @@
+    "runtime/browser/xwalk_presentation_service_helper_win.h",
+    "runtime/browser/xwalk_render_message_filter.cc",
+    "runtime/browser/xwalk_render_message_filter.h",
+    "runtime/browser/xwalk_runner.cc",
+    "runtime/browser/xwalk_runner.h",
+    "runtime/browser/xwalk_runner_win.cc",
+    "runtime/browser/xwalk_runner_win.h",
+    "runtime/browser/xwalk_ssl_host_state_delegate.cc",
+    "runtime/browser/xwalk_ssl_host_state_delegate.h",
+    "runtime/common/android/xwalk_globals_android.cc",
+    "runtime/common/android/xwalk_globals_android.h",
+    "runtime/common/android/xwalk_hit_test_data.cc",
+    "runtime/common/android/xwalk_hit_test_data.h",
+    "runtime/common/android/xwalk_message_generator.cc",
+    "runtime/common/android/xwalk_message_generator.h",
+    "runtime/common/android/xwalk_render_view_messages.cc",
+    "runtime/common/android/xwalk_render_view_messages.h",
+    "runtime/common/logging_xwalk.cc",
+    "runtime/common/logging_xwalk.h",
+    "runtime/common/paths_mac.h",
+    "runtime/common/paths_mac.mm",
+    "runtime/common/xwalk_common_message_generator.cc",
+    "runtime/common/xwalk_common_message_generator.h",
+    "runtime/common/xwalk_common_messages.cc",
+    "runtime/common/xwalk_common_messages.h",
+    "runtime/common/xwalk_content_client.cc",
+    "runtime/common/xwalk_content_client.h",
+    "runtime/common/xwalk_localized_error.cc",
+    "runtime/common/xwalk_localized_error.h",
+    "runtime/common/xwalk_paths.cc",
+    "runtime/common/xwalk_paths.h",
+    "runtime/common/xwalk_resource_delegate.cc",
+    "runtime/common/xwalk_resource_delegate.h",
+    "runtime/common/xwalk_runtime_features.cc",
+    "runtime/common/xwalk_runtime_features.h",
+    "runtime/common/xwalk_switches.cc",
+    "runtime/common/xwalk_switches.h",
+    "runtime/common/xwalk_system_locale.cc",
+    "runtime/common/xwalk_system_locale.h",
+    "runtime/renderer/android/xwalk_permission_client.cc",
+    "runtime/renderer/android/xwalk_permission_client.h",
+    "runtime/renderer/android/xwalk_render_thread_observer.cc",
+    "runtime/renderer/android/xwalk_render_thread_observer.h",
+    "runtime/renderer/android/xwalk_render_view_ext.cc",
+    "runtime/renderer/android/xwalk_render_view_ext.h",
+    "runtime/renderer/isolated_file_system.cc",
+    "runtime/renderer/isolated_file_system.h",
+    "runtime/renderer/xwalk_content_renderer_client.cc",
+    "runtime/renderer/xwalk_content_renderer_client.h",
+    "runtime/renderer/xwalk_render_thread_observer_generic.cc",
+    "runtime/renderer/xwalk_render_thread_observer_generic.h",
+  ]
+  if (is_linux && !is_component_build && is_component_ffmpeg) {
+    # Set RPath for executable xwalk, xwalk_browsertest and xwalk_unittest.
+    public_configs = [ "//build/config/gcc:rpath_for_built_shared_libraries" ]
+  }
+
+  # TODO(heke123): Generate header files into gen/xwalk/grit by putting the
+  # files in the right places in grit. Clean below include_dirs.
+  include_dirs = [
```

Comment:
I'm raising this issue because it's really easy to fix and at the moment you have the worst of both worlds: `config`s for most of those directories in other `BUILD.gn` files + duplication of the same lines here.

* `xwalk_application_resources_include_dir` takes care of adding `$root_gen_dir/xwalk/application`
* `xwalk_sysapps_resources_include_dir` takes care of adding `$root_gen_dir/xwalk/sysapps`

This leaves us with `$root_gen_dir/xwalk/resources`, which can be solved by a config entry in `resources/BUILD.gn` just like the others you already have, and `$root_gen_dir/xwalk`, which is the only one that makes sense here (with a TODO for getting rid of it replacing the includes with proper, specialized ones).